tests/base_test.py

```diff
@@ -214,10 +215,13 @@ def workflow_sync_response(file_name):
     def assert_success(self, rv, msg=""):
         try:
             data = json.loads(rv.get_data(as_text=True))
+            error_message = ""
+            if 'message' in data:
+                error_message = data['message']
             self.assertTrue(200 <= rv.status_code < 300,
                             "BAD Response: %i. \n %s" %
-                            (rv.status_code, data['message']) + ". " + msg + ". ")
-        except:
+                            (rv.status_code, error_message + ". " + msg + ". "))
+        except Exception as e:
             self.assertTrue(200 <= rv.status_code < 300,
                             "BAD Response: %i." % rv.status_code + ". " + msg)
 
```

Comment:
I don't understand what's going on in the except clause.
Why are we doing another assertion in the except clause?

Reply:
agreed, very confusing code.  Re-writing.